### PR TITLE
Implement a real Xcode locator

### PR DESCRIPTION
**Pull request by siggisim**
This removes the need to have hardcoded XCode paths in our code (and for folks running executors to place XCode at those paths), and also allows us to support XCode versions more precise than two components (i.e. 12.5.1 or 12.5.1.12E507).

Xcode location logic is losely based on Bazel's XCode locator: https://github.com/bazelbuild/bazel/blob/master/tools/osx/xcode_locator.m

This removes the need for enableXcodeOverride.

---

**Version bump**: Minor

### CODE REVIEW

File: enterprise/server/cmd/executor/executor.go [hunk 1]
```diff
@@ -114,6 +115,10 @@ func GetConfiguredEnvironmentOrDie(configurator *config.Configurator, healthChec
 		log.Infof("No authentication will be configured: %s", err)
 	}
 
+	xcodeLocator := xcode.NewXcodeLocator()
+	xcodeLocator.Locate()
```
Comment (tylerwilliams) [inline enterprise/server/cmd/executor/executor.go:119]: I think you could have "NewXcodeLocator()" call locate() internally before returning, then remove the "Locate()" method from the interface, and from being a public method

Comment (siggisim) [reply enterprise/server/cmd/executor/executor.go]: Done. Also added support for SDKs, dropped enableXcodeOverride, and simplified the CFString stuff (I think all of that craziness was only needed when allocating a new string - which we don't do).

File: server/xcode/xcode_darwin.go [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+
+		versions := expandXCodeVersions(xcodePlist.CFBundleShortVersionString, xcodePlist.ProductBuildVersion)
+		mostPreciseVersion := versions[len(versions)-1]
+		for _, version := range versions {
+			existingXcode, ok := versionMap[version]
+			if ok && mostPreciseVersion < existingXcode.version {
+				continue
+			}
+			log.Infof("Found XCode version: %s=>%s", version, developerDirPath)
+			log.Debugf("With SDKs %+v", sdks)
```
Comment (tylerwilliams) [inline server/xcode/xcode_darwin.go:130]: can you consolidate this into the line above (or rm it)?

**Review by tylerwilliams — APPROVED** [review]
LGTM (with one logging nit)

File: server/xcode/xcode_darwin.go (hunk 2 above)
Comment (siggisim) [reply server/xcode/xcode_darwin.go]: Done

File: enterprise/server/cmd/executor/executor.go [hunk 3]
```diff
@@ -114,6 +115,8 @@ func GetConfiguredEnvironmentOrDie(configurator *config.Configurator, healthChec
 		log.Infof("No authentication will be configured: %s", err)
 	}
 
+	realEnv.SetXCodeLocator(xcode.NewXcodeLocator())
```
Comment (brentleyjones) [inline enterprise/server/cmd/executor/executor.go:118]: nit: It's `Xcode`, not `XCode`

**Pull request merged**